### PR TITLE
Remove redis from api dependencies

### LINKED ISSUE #2865: Move redis out of api dependencies list
**What would you like to be added**:

Currently, `redis` is counted as web_api and piped_api dependency but we should move it out of those API dependencies list to keep the API logic flexible.

ref: 
https://github.com/pipe-cd/pipecd/blob/master/pkg/app/server/grpcapi/web_api.go#L72
https://github.com/pipe-cd/pipecd/blob/master/pkg/app/server/grpcapi/piped_api.go#L65

**Why is this needed**:

### PR DESCRIPTION
**Pull request by Hosshii**
**What this PR does / why we need it**:
Remove redis from web_api and piped_api dependency.

**Which issue(s) this PR fixes**:

Fixes #2865

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/app/server/grpcapi/web_api.go [hunk 1]
```diff
@@ -121,7 +118,7 @@ type WebAPI struct {
 	pipedProjectCache      cache.Cache
 	pipedStatCache         cache.Cache
 	insightCache           cache.Cache
-	redis                  redis.Redis
+	unregisteredAppStore   unregisteredappstore.Store
```
Comment (khanhtc1202) [inline pkg/app/server/grpcapi/web_api.go:121]: nits, should move this to the upper block (nearby other stores) for better readable code 👍  (same with the piped_api.go PipedApi struct)

**Comment by nghialv** [comment]
@Hosshii I think you need to run `make gazelle` to add missing dependencies.

File: pkg/app/server/grpcapi/piped_api.go [hunk 2]
```diff
@@ -100,14 +97,14 @@ type PipedAPI struct {
...
-	redis                redis.Redis
+	unregisteredAppStore unregisteredappstore.Store
 
 	webBaseURL string
 	logger     *zap.Logger
 }
 
 // NewPipedAPI creates a new PipedAPI instance.
-func NewPipedAPI(ctx context.Context, ds datastore.DataStore, sc cache.Cache, sls stagelogstore.Store, alss applicationlivestatestore.Store, las analysisresultstore.Store, hc cache.Cache, rd redis.Redis, cop commandOutputPutter, webBaseURL string, logger *zap.Logger) *PipedAPI {
+func NewPipedAPI(ctx context.Context, ds datastore.DataStore, sc cache.Cache, sls stagelogstore.Store, alss applicationlivestatestore.Store, las analysisresultstore.Store, hc cache.Cache, uac unregisteredappstore.Store, cop commandOutputPutter, webBaseURL string, logger *zap.Logger) *PipedAPI {
```
Comment (knanao) [inline pkg/app/server/grpcapi/piped_api.go:107]: nits
```suggestion
func NewPipedAPI(ctx context.Context, ds datastore.DataStore, sc cache.Cache, sls stagelogstore.Store, alss applicationlivestatestore.Store, las analysisresultstore.Store, hc cache.Cache, uas unregisteredappstore.Store, cop commandOutputPutter, webBaseURL string, logger *zap.Logger) *PipedAPI {
```

File: pkg/app/server/grpcapi/piped_api.go [hunk 3]
```diff
@@ -123,7 +120,7 @@ func NewPipedAPI(ctx context.Context, ds datastore.DataStore, sc cache.Cache, sl
 		appPipedCache:             memorycache.NewTTLCache(ctx, 24*time.Hour, 3*time.Hour),
 		deploymentPipedCache:      memorycache.NewTTLCache(ctx, 24*time.Hour, 3*time.Hour),
 		pipedStatCache:            hc,
-		redis:                     rd,
+		unregisteredAppStore:      uac,
```
Comment (knanao) [inline pkg/app/server/grpcapi/piped_api.go:123]:
```suggestion
		unregisteredAppStore:      uas,
```

File: pkg/model/BUILD.bazel [hunk 4]
```diff
@@ -55,6 +55,7 @@ go_library(
         "deployment.go",
         "deployment_chain.go",
         "docs.go",
+        "environment.pb.go",
```
Comment (nghialv) [inline pkg/model/BUILD.bazel:58]: Please remove this since it is no longer needed.
(I think you have to delete it at your local as well.)

**Comment by Hosshii** [comment]
Thank you for reviewing. 
I fixed. PTAL

**Comment by nghialv** [comment]
Nice. Keep up your great work!
/lgtm

**Comment by khanhtc1202** [comment]
Nice improvement 👍 
/approve

**Pull request merged**